src/Pay/Adapter/Stripe.php

```diff
@@ -0,0 +1,201 @@
+    {
+        $path = '/customers/' . $customerId . '/sources/' . $cardId;
+        $requestBody = [];
+        if (!empty($name)) {
+            $requestBody['name'] = $name;
+        }
+        if (!empty($expMonth)) {
+            $requestBody['exp_month'] = $expMonth;
+        }
+        if (!empty($expYear)) {
+            $requestBody['exp_year'] = $expYear;
+        }
+        if (!empty($billingDetails)) {
+            $requestBody['address_city'] = $billingDetails['address_city'] ?? null;
+            $requestBody['address_country'] = $billingDetails['address_country'] ?? null;
+            $requestBody['address_line1'] = $billingDetails['address_line1'];
+            $requestBody['address_line2'] = $billingDetails['address_line2'];
+            $requestBody['address_state'] = $billingDetails['address_state'];
+            $requestBody['address_zip'] = $billingDetails['address_zip'];
+        }
+        return $this->execute(self::METHOD_POST, $path, $requestBody);
+    }
+
+    /**
+     * Get a card
+     */
+    public function getCard(string $customerId, string $cardId): array
+    {
+        $path = '/customers/' . $customerId . '/sources/' . $cardId;
+        return $this->execute(self::METHOD_GET, $path);
+    }
+
+    /**
+     * Delete a credit card record
+     */
+    public function deleteCard(string $customerId, string $cardId): bool
+    {
+        $path = '/customers/' . $customerId . '/sources/' . $cardId;
+        $res =  $this->execute(self::METHOD_DELETE, $path);
+        return $res['deleted'] ?? false;
+    }
+
+    /**
+     * Add new customer in the gateway database
+     * returns the newly created customer
+     *
+     * @throws Exception
+     */
+    public function createCustomer(string $name, string $email, array $billingDetails = [], string $paymentMethod = null): array
+    {
+        $path = '/customers';
+        $requestBody = [
+            'name' => $name,
+            'email' => $email,
+        ];
+        if (!empty($paymentMethod)) {
+            $requestBody['payment_method'] = $paymentMethod;
+        }
+        if (!empty($billingDetails)) {
+            $requestBody['billing_details'] = $billingDetails;
+        }
+        $result = $this->execute(self::METHOD_POST, $path, $requestBody);
+        return $result;
+    }
+
+    /**
+     * List customers
+     */
+    public function listCustomers(): array
+    {
+        return $this->execute(self::METHOD_GET, '/customers');
+    }
+
+    /**
+     * Get customer details by ID
+     */
+    public function getCustomer(string $customerId): array
+    {
+        $path = '/customers/' . $customerId;
+        $result = $this->execute(self::METHOD_GET, $path);
+        return $result;
+    }
+
+    /**
+     * Update customer details
+     */
+    public function updateCustomer(string $customerId, string $name, string $email, array $billingDetails = [], string $paymentMethod = null): array
```

Comment:
What does `$billingDetails` param should accept? should we make this an object so we better manage the complex data type? What does `$paymentMethod` values should be?

Reply:
PaymentMethod is 
```
payment_method
optional
The ID of the PaymentMethod to attach to the customer.
```

Reply:
For customer, it should have been address not billing details, billingDetails is no purchase method. I've updated.

We can define PHP class for both address and billing details. Let me know if we want to do that

Reply:
I think it will help us